lib/resty/upstream/healthcheck.lua

```diff
@@ -602,6 +606,16 @@ function _M.spawn_checker(opts)
         concurrency = concur,
     }
 
+    -- as far as we do not support https override ports if set
+    if http_port then
+        for i, _ in ipairs(ctx.primary_peers) do
+            ctx.primary_peers[i].port = http_port
```

Comment:
@sitano As I've said, it's better to add proper https support here instead of overwriting peer ports with brute force.